src/main/java/com/microsoft/aad/msal4j/PublicClientApplication.java

```diff
@@ -55,11 +55,6 @@
     @Override
     public CompletableFuture<IAuthenticationResult> acquireToken(DeviceCodeFlowParameters parameters) {
 
-        if (!AuthorityType.AAD.equals(authenticationAuthority.authorityType())) {
-            throw new IllegalArgumentException(
-                    "Invalid authority type. Device Flow is only supported by AAD authority");
-        }
-
```

Comment:
Is device code flow supported by B2C Authority ?

Reply:
B2C does not support device code flow. We should check if authority is B2C, and then throw error if so

Reply:
For some reason I originally read that as "if ADFS" rather than "if not AD", so I incorrectly got rid of the logic for blocking B2C. I re-added the authority type check and exception throw, so that it now throws the exception if the given authority is not AD or ADFS, and also re-add the related unit test.